hs/src/Reach/JSUtil.hs

```diff
@@ -267,19 +320,63 @@ instance HasJSAnnot JSExpression where
     JSVarInitExpression a _ -> jsa a
     JSYieldExpression a _ -> a
     JSYieldFromExpression a _ _ -> a
+  rjsa a' = \case
+    JSIdentifier _ x -> JSIdentifier a' x
+    JSDecimal _ x -> JSDecimal a' x
+    JSLiteral _ x -> JSLiteral a' x
+    JSHexInteger _ x -> JSHexInteger a' x
+    JSOctal _ x -> JSOctal a' x
+    JSStringLiteral _ x -> JSStringLiteral a' x
+    JSRegEx _ x -> JSRegEx a' x
+    JSArrayLiteral _ x _ -> JSArrayLiteral a' x a'
+    JSAssignExpression a x y -> JSAssignExpression (rjsa a' a) (rjsa a' x) (rjsa a' y)
+    JSAwaitExpression _ x -> JSAwaitExpression a' (rjsa a' x)
+    JSCallExpression a _ y _ -> JSCallExpression (rjsa a' a) a' (rjsaJSL a' y) a'
+    JSCallExpressionDot a _ y -> JSCallExpressionDot (rjsa a' a) a' (rjsa a' y)
+    JSCallExpressionSquare a _ y _ -> JSCallExpressionSquare (rjsa a' a) a' (rjsa a' y) a'
+    JSClassExpression _ x y z r _ -> JSClassExpression a' x y z r a'
+    JSCommaExpression a _ y -> JSCommaExpression (rjsa a' a) a' (rjsa a' y)
+    JSExpressionBinary a x y -> JSExpressionBinary (rjsa a' a) (rjsa a' x) (rjsa a' y)
+    JSExpressionParen _ x _ -> JSExpressionParen a' (rjsa a' x) a'
+    JSExpressionPostfix a x -> JSExpressionPostfix (rjsa a' a) (rjsa a' x)
+    JSExpressionTernary a _ y _ r -> JSExpressionTernary (rjsa a' a) a' (rjsa a' y) a' (rjsa a' r)
+    JSArrowExpression x _ y -> JSArrowExpression x a' y
+    JSFunctionExpression _ x _ z _ s -> JSFunctionExpression a' x a' (rjsaJSL a' z) a' s
+    JSGeneratorExpression _ _ y _ r _ t -> JSGeneratorExpression a' a' y a' (rjsaJSL a' r) a' t
+    JSMemberDot a _ y -> JSMemberDot (rjsa a' a) a' (rjsa a' y)
+    JSMemberExpression a _ y _ -> JSMemberExpression (rjsa a' a) a' (rjsaJSL a' y) a'
+    JSMemberNew _ x _ z _ -> JSMemberNew a' (rjsa a' x) a' (rjsaJSL a' z) a'
+    JSMemberSquare a _ y _ -> JSMemberSquare (rjsa a' a) a'  (rjsa a' y) a'
+    JSNewExpression _ x -> JSNewExpression a' (rjsa a' x)
+    JSObjectLiteral _ x _ -> JSObjectLiteral a' (mkCommaTrailingList (map (rjsa a') $ jso_flatten x)) a'
+    JSSpreadExpression _ x -> JSSpreadExpression a' (rjsa a' x)
+    JSTemplateLiteral x _ y z -> JSTemplateLiteral (fmap (rjsa a') x) a' y z
+    JSUnaryExpression a x -> JSUnaryExpression (rjsa a' a) (rjsa a' x)
+    JSVarInitExpression a x -> JSVarInitExpression (rjsa a' a) x
```

Comment:
x is important to look at

Reply:
why isn't x currently a part of this typeclass then?

Reply:
Cause I'm lazy :)